pyInterface/testMC.sh

```diff
@@ -0,0 +1,190 @@
+
+cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/generator_noBeamSimulation.conf" "./"
+cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/template.key" "./"
+cp "${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/wavelist.compass.2008.88waves" "./"
+cp "${ROOTPWA}/pyInterface/rootpwa.config" "./"
+cp "${ROOTPWA}/pyInterface/mcTest/reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root.example" "./reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root"
+
+#-- END PREPARATION --#
+
+### BEGIN MONTE CARLO GENERATION ###
+
+# generate phase space data
+echo "Generating phase space ..."
+if ! ${ROOTPWA}/build/bin/genpw.py -n $NMB_PS_EVENTS -p "${PARTICLE_DATA_TABLE}" -M $MASS -B $BINWIDTH "./generator_noBeamSimulation.conf" -o "./data/phase_space_MASS_${MASS}-$((MASS+BINWIDTH))_N_${NMB_PS_EVENTS}.root"; then
+	echo "Generation of phase space was not successful! Aborting."
+	exit 1
+fi
+
+# generate the keyfiles
+echo "Generating keyfiles ..."
+DESTINATION_DIR="${TESTDIR}/keyfiles"
+export PARTICLE_DATA_TABLE
+export DESTINATION_DIR
+export WAVESET_FILES="wavelist.compass.2008.88waves"
+if ! ${ROOTPWA}/userAnalysisWorkspace/3pi.--+/keyfiles/GenerateKeyfiles.sh; then
+	echo "Generation of keyfiles was not successful! Aborting."
+	exit 1
+fi
+
+# remove some parameterizations
+rm ${TESTDIR}/keyfiles/*[f0_980_0bw=[*.key
+rm ${TESTDIR}/keyfiles/*[f0_980_0=[*.key
+
+# create first file manager
+echo "Create first file manager ..."
+if ! ${ROOTPWA}/build/bin/createFileManager.py; then
+	echo "Creation of first file manager was not successful! Aborting."
+	exit 1
+fi
+
+# calculate amplitudes for generated data
+echo "Calculate amplitudes for generated data ..."
+if ! ${ROOTPWA}/build/bin/calcAmplitudes.py -e generated; then
+	echo "Calculation of amplitudes for generated data was not successful! Aborting."
+	exit 1
+fi
+
+# calculate integrals for generated data
+echo "Calculate integrals for generated data ..."
+if ! ${ROOTPWA}/build/bin/calcIntegrals.py -e generated; then
+	echo "Calculation of integrals for generated was not successful! Aborting."
+	exit 1
+fi
+
+# generate weighted MC pseudo data
+echo "Generate weighted MC pseudo data ..."
+if ! ${ROOTPWA}/build/bin/genPseudoData.py "./generator_noBeamSimulation.conf" "${TESTDIR}/reference_fit/bin65_c2pap_bestfits_converged_MASS_1800_1820_N45340.root" "./ints/integral_binID-0_2.root" "./weighted_mc_data/weighted_pseudoData_MASS_${MASS}-$((MASS+BINWIDTH))_N_${NMB_PSEUDO_EVENTS}.root" -s 654321 -n ${NMB_PSEUDO_EVENTS} -M ${MASS} -B ${BINWIDTH}; then 
+	echo "Generation of weighted MC pseudo data was not successful! Aborting."
+	exit 1
+fi
+
+# deweight the pseudo data
+echo "Deweight pseudo data ..."
+if ! ${ROOTPWA}/build/bin/deWeight.py "./weighted_mc_data/weighted_pseudoData_MASS_${MASS}-$((MASS+BINWIDTH))_N_${NMB_PSEUDO_EVENTS}.root" "./data/pseudoData_MASS_${MASS}-$((MASS+BINWIDTH))_N_${NMB_PSEUDO_EVENTS}.root"; then
```

Comment:
you could specify a constant seed. at the moment the default seed of `deWeight.py` is a constant number not equal to zero so this would not change anything right now, but might be more future-proof.